src/main/java/org/mockito/internal/configuration/MockAnnotationProcessor.java

```diff
@@ -41,6 +50,29 @@ public static Object processAnnotationForMock(Mock annotation, Class<?> type, St
 
         // see @Mock answer default value
         mockSettings.defaultAnswer(annotation.answer());
-        return Mockito.mock(type, mockSettings);
+
+        if (type == MockedStatic.class) {
+            return Mockito.mockStatic(inferStaticMock(genericType.get(), name), mockSettings);
+        } else {
+            return Mockito.mock(type, mockSettings);
+        }
+    }
+
+    private static Class<?> inferStaticMock(Type type, String name) {
+        if (type instanceof ParameterizedType) {
+            ParameterizedType parameterizedType = (ParameterizedType) type;
+            return (Class<?>) parameterizedType.getRawType();
```

Comment:
Shouldn't this use `getActualTypeArguments()[0]` instead?
It's failing for me now since `getRawType()` returns `MockedStatic`

Reply:
Do you mind opening a PR with regression test + fix? Thanks!

Reply:
Hmm, seems this was easy to fix locally with my suggestion, but I now get another error, which I don't know why it happens:
```
org.mockito.exceptions.misusing.NotAMockException: Argument passed to Mockito.mockingDetails() should be a mock, but is an instance of class java.lang.Class!

	at org.mockito.junit.jupiter.MockitoExtension.afterEach(MockitoExtension.java:186)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeAfterEachCallbacks$11(TestMethodTestDescriptor.java:255)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor$$Lambda$316/0000000000000000.invoke(Unknown Source)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeAllAfterMethodsOrCallbacks$12(TestMethodTestDescriptor.java:271)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor$$Lambda$1809/0000000000000000.execute(Unknown Source)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeAllAfterMethodsOrCallbacks$13(TestMethodTestDescriptor.java:271)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor$$Lambda$314/0000000000000000.accept(Unknown Source)
```
The class with the static method ends up there...

Reply:
I isolated the error. The problem is that the mocks for static methods are scoped. This means that they are released at the end of the try-with-resources block what implies that the mocks are no longer valid at the end of the method resulting in this error. 

@TimvdLippe For now I think the easiest is to exclude static mocks from the post processing. The more correct approach would be to execute the "after each" logic upon releasing the mock rather then at the end of the test. I will look into it but need to understand the post-processing first.

Reply:
Btw, my setup was with and without a try-with-resources block, since I understood from the documentation of `@Mock` that it should work with that as well, which would simplify things (and since I already had other fields that were annotated with `@Mock`)

Reply:
```java
@ExtendWith(MockitoExtension.class)
public class TenantScopeTest {
    @Mock
    private UserPrincipal principal;

    @Mock
    private MockedStatic<UserPrincipalUtil> userPrincipalUtil;

    @Test
    public void opsShouldBeCachedInstancesByNameAndTenant() {
        String tenant = "x";
        when(principal.getMainTenant()).thenReturn(tenant);

        userPrincipalUtil.when(UserPrincipalUtil::requireUserPrincipal).thenReturn(principal);

        //....
}
```

Reply:
@andrei-ivanov I have not had the time to test it yet (we should have unit tests for the runners using both mock makers) but maybe the 'runner-fix' branch already fixes the problem.

Reply:
I still get the `NotAMockException` with the `runner-fix` branch

Reply:
Let me know when you want to do run some more tests, I'm eager to get rid of PowerMock 😁

Reply:
I also have this problem, when the mockMaker is InlineByteBuddyMockMaker, I use SpyBean to spy a spring bean, and then verify the bean, it throws NotAMockException. I debug and find the class type is not contains mockito, but it contains when I use the ByteBuddyMockMaker.